docs/tutorials/guides/how-to-compile-smart-contract.md

```diff
@@ -0,0 +1,188 @@
+---
+title: 'How to Create and Compile Smart Contracts with Docker'
+order: -99982
+outline: [0, 5]
+---
+
+# How to Create and Compile Smart Contracts with Docker
+This document guides you on how to create, compile, and deploy smart contracts using Ultra's 3rd party Docker image.
+
+## Prerequisites
+Before you begin, ensure that Docker is installed on your computer.
+
+## Step 1: Pull the Ultra Image
+Pull the Ultra 3rd party Docker image with the following command:
+
+```sh
+docker pull quay.io/ultra.io/3rdparty-devtools
+```
+
+Verify the Ultra image with the following command:
+```sh
+docker image ls quay.io/ultra.io/3rdparty-devtools
+```
+
+![](./images/compile-smart-contract-pull-image.png)
+## Step 2: Start the Docker Container
+Start the Docker container, which contains the smart contract developer tools and a local Ultra Blockchain, using the following command:
+
+```sh
+docker run -dit --name ultra -p 8888:8888 -p 9876:9876 -v ~/ultra_workdir:/opt/ultra_workdir quay.io/ultra.io/3rdparty-devtools
+```
+
+Verify that the Ultra Docker container is up and running with:
+
+```sh
+docker ps -a
+```
+![](./images/compile-smart-contract-start-docker.png)
+
+## Step 3: Start the Local Chain in the Docker Container
+Log in to the container with:
+
+```sh
+docker exec -it ultra /bin/bash
+```
+
+Start the local Ultra Blockchain with:
+
+```sh
+ultratest -D -n -s
+```
+Accept the default recommendation for the temporary files location by pressing enter.
+
+![](./images/compile-smart-contract-start-local-chain.png)
+
+### Step 3a: Verify the Local Chain is Running
+Execute the following command inside the container:
+
+```sh
+cleos get info
+```
+
+Or from outside the container on the host machine:
+```sh
+docker exec ultra cleos get info
+```
+![](./images/compile-smart-contract-verfy-local-chain.png)
+
+## Step 4: Create and Compile the Smart Contract
+Create a directory on your host machine in `ultra_workdir/hello` and create a file named `hello.cpp` with the following content:
```

Comment:
Maybe having a screenshot showing the folder structure will help a bit. To be clear that this is on host machine.

And just clarify that `hello.cpp` is just a text file